src/common/snippets/src/pass/mlp_seq_tokenization.cpp

```diff
@@ -0,0 +1,204 @@
+         *               |
+         *              ...
+         */
+
+        const auto matmul0 = ov::as_type_ptr<ov::opset1::MatMul>(pattern_to_output.at(m_matmul0).get_node_shared_ptr());
+        if (!is_matmul_supported(matmul0) || !has_one_consumer(matmul0)) {
+            return false;
+        }
+
+        if (transformation_callback(matmul0)) {
+            return false;
+        }
+
+        // Add possible FQ before matmul0
+        if (const auto fq = ov::as_type_ptr<ov::op::v0::FakeQuantize>(matmul0->input_value(0).get_node_shared_ptr())) {
+            if (has_one_consumer(fq)) {
+                potential_body_params_count += ov::snippets::utils::get_non_scalar_constant_count_for_fq(fq);
+                ordered_ops.push_back(fq);
+            }
+        }
+        ordered_ops.push_back(matmul0);
+
+        // Tokenize Sequence while we can do it
+        std::shared_ptr<ov::Node> prev_op = matmul0;
+        auto interm_op = prev_op->get_output_target_inputs(0).begin()->get_node()->shared_from_this();
+        auto available_regs = config.get_data_ptr_gpr_count();
+
+        while (has_one_consumer(prev_op)) {
+            auto current_potential_body_params_count = potential_body_params_count;
+
+            if (is_matmul_supported(interm_op) && !transformation_callback(interm_op)) {
+                // +1 for weights
+                current_potential_body_params_count++;
+            } else if (is_supported_intermediate_op(interm_op)) {
+                current_potential_body_params_count += get_potential_body_params(interm_op);
+            } else {
+                // Unsupported op
+                break;
+            }
+
+            // TODO [75567]: move this plugin-specific constraint to the plugin callback
+            // +1 - for result
+            if (current_potential_body_params_count + unique_buffer_reg_group_count + 1 > available_regs)
+                break;
+
+            ordered_ops.push_back(interm_op);
+            prev_op = interm_op;
+            interm_op = prev_op->get_output_target_inputs(0).begin()->get_node()->shared_from_this();
+
+            // Move counts
+            potential_body_params_count = current_potential_body_params_count;
+        }
+
+        // Currently, sequence of MLP should contain 2 MatMuls at least
+        size_t mm_count = 0;
+        for (const auto& op : ordered_ops) {
+            mm_count += (ov::is_type<ov::op::v0::MatMul>(op));
+        }
+        if (mm_count < 2)
+            return false;
+
+        /* ================================ */
+
+        /* ====== Subgraph creation ======= */
+
+        const auto subgraph = ov::snippets::utils::tokenize_ordered_nodes(ordered_ops);
+
+        // mark the Subgraph as Completed to not allow Snippets to include any nodes into this Subgraph in common Tokenization
+        SetSnippetsSubgraphType(subgraph, SnippetsSubgraphType::Completed);
+
+        return true;
+
+        /* ================================ */
```

Comment:
I suppose we can remove `/* ... */` comments since now subgraph tokenization part takes only one line: it's already easy to read and understand

Reply:
Removed